deployment/terraform/cdn.tf

```diff
@@ -17,16 +16,56 @@ resource "aws_cloudfront_distribution" "tilegarden" {
     }
   }
 
+  origin {
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      # S3 websites don't support TLS :/
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2", "TLSv1.1", "TLSv1"]
+    }
+
+    domain_name = "${aws_s3_bucket.tile_cache.website_endpoint}"
+    origin_id   = "tilegardenCacheOriginEastId"
+
+    custom_header {
+      name  = "Accept"
+      value = "image/png"
+    }
+  }
+
   aliases         = ["tiles.${var.r53_public_hosted_zone}"]
   price_class     = "${var.cloudfront_price_class}"
   enabled         = true
   is_ipv6_enabled = true
   comment         = "${var.project} (${var.environment})"
 
+  # Tilegarden cache origin
   default_cache_behavior {
+    allowed_methods  = ["GET", "HEAD", "OPTIONS"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = "tilegardenCacheOriginEastId"
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = "300"               # Five minutes
```

Comment:
When I was testing out staging, I got some tiles that timed out, then the error message was cached for 5 minutes even though the Lambda function had actually succeeded and written the tile to S3.

So that points to one issue that should be addressed, I assume on the origin definition--the "Origin Response Timeout" should be no shorter than the timeout on the Lambda function (currently 1 minute).

I also think it makes sense to set the TTL for the "new tiles" endpoint as short as is reasonable, so that if anything goes wrong it won't be as long until it tries again.  There might not actually be a period during which serving a cached response on the API Gateway origin saves a Lambda invocation--before the first one finishes there's no cached response to return, and the tile gets written to S3 before it gets returned from the Lambda function, so there might not be a time during which a request to the S3 cache would miss but one to the API Gateway origin would hit.

So 5 minutes seems fine for the S3 origin (longer would probably be fine, too, but I don't know if there would be any advantage to it) but it should be overridden for the "new tiles" cache_behavior below, probably to also be 1 minute or slightly more.

Reply:
I'm glad you caught this, very subtle but important behavior here. I'll bump the Origin Read Timeout to 60 seconds, which should hopefully make timeouts less common.

I agree, I don't see any benefit to caching the Lambda origin. Can you think of any downside to just setting `min_ttl`, `default_ttl`, and `max_ttl` to `0` for that origin?

Reply:
> Can you think of any downside to just setting min_ttl, default_ttl, and max_ttl to 0 for that origin?

Actually now I'm thinking I might have missed an interaction in what I wrote above--would the redirect from `/` to `/latest/` be cached using the default TTL from `default_cache_behavior`?  I.e. having gotten a redirect on the initial attempt, will it keep redirecting until the timeout has passed?  I was thinking the cached error was entirely the fault of the cache rule on the Lambda endpoint, but now I'm thinking I was getting two cached responses.  In which case having a longer TTL on the root/S3 endpoint than on the Lambda endpoint would mean extra invocations until the cached redirect expires.

Reply:
> I.e. having gotten a redirect on the initial attempt, will it keep redirecting until the timeout has passed?

Good point, I think it will indeed cache the redirect -- see my note in the testing instructions above. Sounds like the TTLs should be identical for both origins.

Reply:
Yeah, seems like.  The change to `origin_read_timeout` should take care of the original issue I saw, though, so hopefully there won't be any caching of bad responses anyway.